auth-request.lua

```diff
@@ -89,33 +139,57 @@ core.register_action("auth-request", { "http-req" }, function(txn, be, path)
 	end
 
 	-- Make request to backend.
-	local response, err = http.head {
+	if method == "*" then
+		method = txn.sf:method()
+	end
+	local response, err = http.send(method:upper(), {
 		url = "http://" .. addr .. path,
 		headers = headers,
-	}
+	})
+
+	-- `terminate_on_failure == true` means that the Lua script should send the response
+	-- and terminate the transaction in the case of a failure. This will happen when
+	-- hdr_fail content isn't a dash `-`.
+	local terminate_on_failure = hdr_fail ~= "-"
 
 	-- Check whether we received a valid HTTP response.
 	if response == nil then
 		txn:Warning("Failure in auth-request backend '" .. be .. "': " .. err)
-		set_var(txn, "txn.auth_response_code", 500)
+		if terminate_on_failure then
+			send_response(txn)
+		else
+			set_var(txn, "txn.auth_response_code", 500)
+		end
 		return
 	end
 
-	set_var(txn, "txn.auth_response_code", response.status_code)
+	local response_ok = 200 <= response.status_code and response.status_code < 300
 
-	for header, value in response:get_headers(true) do
-		set_var(txn, "req.auth_response_header." .. sanitize_header_for_variable(header), value)
+	-- Transaction is terminated if response is not 2xx (an authentication failure) and `terminate_on_failure` is true.
+	-- Only need to update variables or add new headers if the transaction will not be terminated.
+	if response_ok or not terminate_on_failure then
+		set_var(txn, "txn.auth_response_code", response.status_code)
+		for header, value in response:get_headers(true) do
+			if hdr_succeed == "-" then
+				set_var(txn, "req.auth_response_header." .. sanitize_header_for_variable(header), value)
+			elseif header_match(header, hdr_succeed) then
+				txn.http:req_set_header(header, value)
+			end
+		end
 	end
```

Comment:
Likewise I would set the variables unconditionally here and only guard the `txn.http:req_set_header` by `response_ok and hdr_succeed ~= "-"`